examples/demo/src/products/CreateRelatedReviewButton.tsx

```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+import { CreateButton, useRecordContext } from 'react-admin';
+
+const CreateRelatedReviewButton = () => {
+    const record = useRecordContext();
+
+    return (
+        <CreateButton
+            to={{
+                pathname: '/reviews/create',
+            }}
```

Comment:
Rather than setting the `to` prop, you should set the `resource` prop and let the button manage the `to` itself.